kevm-pyk/src/kevm_pyk/kdist/plugin.py

```diff
@@ -32,7 +32,6 @@ def build(self, output_dir: Path, deps: dict[str, Path], args: dict[str, Any]) -
             output_dir=output_dir,
             enable_llvm_debug=enable_llvm_debug,
             verbose=verbose,
-            plugin_dir=deps.get('plugin'),
```

Comment:
Letting the framework inject the path for `plugin` has a few advantages:
* it allows for flexibility in target location, e.g. in case we want to add a local build feature
* without it, there's a conceptual circular dependency: `kompile -> kdist -> plugin -> kompile`

Reply:
The disadvantage is that `run_kompile` then becomes KEVM specific, instead of semantics generic. I could move `run_kompile` into a different module if it would break the dependency?

Reply:
Though I guess the calculation of `ccopts` in `run_kompile` is semantics specific...

Reply:
I've made it so that `ccopts` is passed into all calls to `run_kompile` instead, so that `run_kompile` is semantics-generic, and `plugin_dir` is calculated outside that function.

Reply:
`run_kompile` (originally `kevm_kompile`) was meant to be specific to `evm-semantics`. It's a thin wrapper around  the semantics-generic `kompile` that consolidates kompilation logic by computing some arguments (like `md_selector` and `ccopts`) and passing them along.

I agree that semantic-specific logic should be kept to the minimum, but we have to calculate these arguments somewhere. I prefer a single location, which this function was intended to be.

Reply:
Yeah, but there is no need now, the amount of semantics specific stuff is so small (just the location of plugin and thus ccopts), that this `run_kompile` should be upstreamed into pyk even.

I can open a PR doing that, and then refactor this to depend on that one.

Reply:
What's the logic that you'd like to upstream? Kompilation for the Booster? I don't have anything against that, but I'd still like to have a function that does what this function is aimed to do so that we don't have to pass the same `md_selector`, `ccopts`, etc. in:
* `kevm-pyk` / `kevm_pyk.__main__`
* `kevm-pyk` / `kevm_pyk.kdist.plugin`
* `kevm-pyk` / `tests.integration.test_prove.py`
* `kontrol` / `kontrol.kompile`
* `kontrol` / `tests.integration.test_solc_to_k`